### PR TITLE
New test dist

### PR DESCRIPTION
**Pull request by psav**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 079a6a438addb05f7c984ab1ed039b2fd8fcb346:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c61533a9d8177b2a4532d49fed90780ae5a5f794:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: fixtures/parallelizer/__init__.py [hunk 1]
```diff
@@ -192,6 +193,15 @@ def __init__(self, config):
         self.slave_urls = SlaveDict()
         self.slave_tests = defaultdict(set)
         self.test_groups = self._test_item_generator()
+
+        self._pool = []
+        self.pool_lock = Lock()
+        from utils.conf import cfme_data
+        self.provs = sorted(set(cfme_data['management_systems'].keys()),
+                            key=lambda l: len(l), reverse=True)
```
Comment (mfalesni) [inline fixtures/parallelizer/__init__.py]:
```python
key=len
```

Comment (psav) [inline fixtures/parallelizer/__init__.py]: hahaha

File: fixtures/parallelizer/__init__.py [hunk 2]
```diff
@@ -681,6 +694,80 @@ def _modscope_id_splitter(self, module_items):
...
+            if not self._pool:
+                for test_group in self.test_groups:
+                    self._pool.append(test_group)
+                    for test in test_group:
+                        if '[' in test:
+                            found_prov = []
+                            for pv in self.provs:
+                                if pv in test:
+                                    found_prov.append(pv)
+                                    break
```
Comment (mfalesni) [inline fixtures/parallelizer/__init__.py]: So this finds only the first one and then breaks to prevent confusion of eg. `nested-rhevm` and `rhevm`?

**Comment by dajoRH** [comment]
Lint report for commit 8893bc57306d5f037eb8408a9c6dddf89abbffd1:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: fixtures/parallelizer/__init__.py (hunk 2 above)
Comment (psav) [inline fixtures/parallelizer/__init__.py]: yes

**Comment by dajoRH** [comment]
Lint report for commit 0260e27b22def136bf440b6b47352197a71dd9df:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajohnso** [comment]
This one looks ready to merge, whatcha think @psav @mfalesni , maybe wait until the end of the week once the beta goes out the door?

**Comment by psav** [comment]
Sounds good!

**Comment by mfalesni** [comment]
Looks good, merging.

**Pull request merged**